src/components/PrivateConsumerAccessFields.tsx

```diff
@@ -0,0 +1,508 @@
+  }
+`;
+
+const PrivateConsumerAccessFields = () => {
+  const { t } = useTranslation();
+  const user = useSelector((state: RootState) => state.user);
+  const { values } = useFormikContext<Resource>();
+  const [privateAccessList, setPrivateAccessList] = useState<ResourceReadAccess[]>([]);
+  const [updatingPrivateAccessList, setUpdatingPrivateAccessList] = useState(false);
+  const [showAddAccessPopover, setShowAddAccessPopover] = useState(false);
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const [showPersonAccessField, setShowPersonAccessField] = useState(false);
+  const [personAccessTextFieldValue, setPersonAccessFieldTextValue] = useState('');
+  const [personAccessTextFieldValueError, setPersonAccessTextFieldValueError] = useState(false);
+  const [savePrivateAccessNetworkError, setSavePrivateAccessNetworkError] = useState(false);
+  const [courses, setCourses] = useState<Course[]>([]);
+  const [waitingForCourses, setWaitingForCourses] = useState(false);
+  const [showCourseAutoComplete, setShowCourseAutocomplete] = useState(false);
+  const [courseAutocompleteValue, setCourseAutocompleteValue] = useState<Course | null>();
+  const [courseAutocompleteTypedValue, setCourseAutocompleteTypedValue] = useState('');
+
+  const addInstitutionPrivateConsumerAccess = async () => {
+    await postCurrentUserInstitutionConsumerAccess(values.identifier);
+    if (privateAccessList.findIndex((privateAccess) => privateAccess.subject === user.institution) === -1) {
+      setPrivateAccessList((prevState) => [
+        ...prevState,
+        {
+          subject: user.institution,
+          profiles: [{ name: ResourceReadAccessNames.Institution }],
+        },
+      ]);
+    }
+  };
+
+  useEffect(() => {
+    const getPrivateAccessList = async () => {
+      const resourceReadAccessListResponse = await getResourceReaders(values.identifier);
+      setPrivateAccessList(resourceReadAccessListResponse.data);
+    };
+    getPrivateAccessList();
+  }, [values.identifier]);
+
+  const deleteAccess = async (access: ResourceReadAccess) => {
+    try {
+      let deleteAPISuccessful = false;
+      if (access.profiles[0].name === ResourceReadAccessNames.Institution) {
+        setUpdatingPrivateAccessList(true);
+        await deleteCurrentUserInstitutionConsumerAccess(values.identifier);
+        deleteAPISuccessful = true;
+      } else if (access.profiles[0].name === ResourceReadAccessNames.Person) {
+        setUpdatingPrivateAccessList(true);
+        await deleteAdditionalUserConsumerAccess(values.identifier, access.subject);
+        deleteAPISuccessful = true;
+      } else if (access.profiles[0].name === ResourceReadAccessNames.Course) {
+        setUpdatingPrivateAccessList(true);
+        const tempCourse = access.subject.split(':: ');
+        deleteAPISuccessful = true;
+        //tempCourse[0]: courseCode, tempCourse[1]: institution, tempCourse[2]: year, tempCourse[3]: Season
```

Comment:
Hadde det gått å gjort om til course-objekt for lesbarheten ? eller blir det for mye fram og tilbake ?

Reply:
Fikset